### DISCUSSION TITLE
How to to get obs variable back into R chunk as a R variable ?

### DISCUSSION THREAD

**Discussion by GuillaumeChretien** [quarto-dev/quarto-cli#9412] [Q&A] [answered]
### Description

How to to get obs variable back into R chunk as a R variable ? I Know ojs_define to transfer an R variable to an ojs variable but how to do the way back ?

**Comment by mcanouil** [comment]
You can't and it's not how it works.
Observable code is executed in the browser and data are stored in the document. This is why you can pass "from R" to Observable.
You don't get to access what's computed in the browser from an R session. Because the R session is long gone at that time.
Even, if you use webR, the processes are different as far as I know.

**Reply by cscheid** [reply, marked as answer]
You can do it if it's a Shiny reactive:

https://quarto.org/docs/interactive/ojs/shiny.html

**Reply by GuillaumeChretien** [reply]
Thanks, just see your answer, that was the exemple i had in mind but wonder if that was my case

**Comment by GuillaumeChretien** [comment]
Thanks ! I thought that with some tricks- maybe shiny context like this exemple https://github.com/quarto-dev/quarto-web/blob/main/docs/interactive/ojs/examples/_shiny/kmeans/kmeans-shiny-ojs.qmd we could use ojs value in R chunck in some ways... my need was to make a maplibre js map (works like a charm in ojs) and bring back the user click information and use the power of R chunk. I'll manage to go on with ojs arquero and js chart librairie. Many tanks, quarto is a great tool !
